CMakeLists.txt

```
@@ -3,16 +3,46 @@ set(CMAKE_CXX_STANDARD 11)
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 
 project(ners570_SpMV CXX)
+message("C++ compiler flags: ${CMAKE_CXX_FLAGS}")
 
 set(SpMV_MAJOR_VERSION 0)
 set(SpMV_MINOR_VERSION 0)
 set(SpMV_PATCH_VERSION 0)
 
 set(SpMV_VERSION ${SpMV_MAJOR_VERSION}.${SpMV_MINOR_VERSION}.${SpMV_PATCH_VERSION})
 
-
 add_subdirectory(src)
 
 enable_testing()
 add_subdirectory(test)
 
+list(APPEND flags "-fPIC" "-Wall")
+if(NOT WIN32)
+  list(APPEND flags "-Wextra" "-Wpedantic")
+endif()
+
+set(CMAKE_CXX_FLAGS         "${CMAKE_CXX_FLAGS} -Wall"
+set(CMAKE_CXX_FLAGS_DEBUG   "-g -traceback")
+set(CMAKE_CXX_FLAGS_RELEASE "-03 -ip xHOST")
+
```

I think maybe we can set the default one (Release for example) in this way:
```suggestion
if(NOT CMAKE_BUILD_TYPE)
    set(CMAKE_BUILD_TYPE "Release")
endif()
```